tests/steps/given.ts

```diff
@@ -19,19 +19,76 @@ Given('I insert GPS data into the database', async function () {
         speed: 30,
         heading: 45,
     };
-
     await GPS.create(gpsData);
 });
 
+Given('I insert AISShips data into the database', async function () {
+    const aisshipData = {
+        ships: [
+          {
+            id: 1,
+            latitude: 12.3323,
+            longitude: 40.2313,
+            speed: 24,
+            heading: 42,
+          },
+          {
+            id: 2,
+            latitude: 13.4456,
+            longitude: 41.1122,
+            speed: 18,
+            heading: 120,
+          },
+          {
+            id: 3,
+            latitude: 11.7890,
+            longitude: 39.5678,
+            speed: 30,
+            heading: 275,
+          },
+        ],
+      };
+    await AISShips.create(aisshipData);
+});
+
 Then('the response data matches the data in the database', async function () {
-    databaseData = await GPS.find({}).then(function(gps) {
+
+    let apiResponseData_GPS;
+    let databaseData_GPS;
+
+    databaseData_GPS = await GPS.find({}).then(function(gps) {
         let transformedGPS = gps.map((data) => data.toJSON())
         return transformedGPS;
     });
 
-    apiResponseData = api.response.data.data[0];
-    expect(apiResponseData.latitude).to.equal(databaseData[0].latitude, 'Data in the response does not match data in the database');
-    expect(apiResponseData.longitude).to.equal(databaseData[0].longitude, 'Data in the response does not match data in the database');
-    expect(apiResponseData.speed).to.equal(databaseData[0].speed, 'Data in the response does not match data in the database');
-    expect(apiResponseData.heading).to.equal(databaseData[0].heading, 'Data in the response does not match data in the database');
+    apiResponseData_GPS = api.response.data.data[0];
+
+    const propertiesToCompare = Object.keys(apiResponseData_GPS);
+    propertiesToCompare.shift();
+    propertiesToCompare.pop();
```

Comment:
Why do you need to shift and pop here?

Reply:
Before the shift and pop, propertiesToCompare = [ '_id', 'latitude', 'longitude', 'speed', 'heading', '__v' ], which becomes [ 'latitude', 'longitude', 'speed', 'heading' ] after the shift and pop. I did this so we would only have to compare latitude, longitude, speed, and heading.

Reply:
Otherwise in the for (const property of propertiesToCompare), there would be an error since it would try to compare id

Reply:
Ahh, I see. I agree we shouldn't compare these values because the fields `_id` and `_v` are used internally by the MongoDB driver and aren't relevant to the external user at all.  

This makes me think that we shouldn't return these fields in the API response at all. Could you please change the query in the AISShips/GPS API so that it specifically excludes these fields? 

This should help: https://stackoverflow.com/questions/33799375/how-to-remove-mongo-specific-fields-from-result-nodejs-mongoose

Once you change this, you shouldn't have to `pop()` and `shift()` either